extensions/amp-apester-media/0.1/monetization/companion/display.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {createElementWithAttributes} from '../../../../../src/dom';
+import {getValueForExpr} from '../../../../../src/json';
+const ALLOWED_AD_PROVIDER = 'gdt';
+import {dict} from '../../../../../src/utils/object';
+
+/**
+ * @param {!JsonObject} media
+ * @param {AmpElement} apesterElement
+ */
+export function handleCompanionDisplay(media, apesterElement) {
+  const companionOptions = getValueForExpr(
+    /**@type {!JsonObject}*/ (media),
+    'campaignData.companionOptions'
+  );
+  if (companionOptions) {
+    const enabledDisplayAd = getValueForExpr(
+      /**@type {!JsonObject}*/ (companionOptions),
+      'enabled'
+    );
+    const settings = getValueForExpr(
+      /**@type {!JsonObject}*/ (companionOptions),
+      'settings'
+    );
+
+    if (
+      enabledDisplayAd &&
+      settings &&
+      settings['bannerAdProvider'] === ALLOWED_AD_PROVIDER
+    ) {
+      const slot = settings['slot'];
+      const defaultBannerSizes = [[300, 250]];
+      const bannerSizes = settings['bannerSizes'] || defaultBannerSizes;
+      constructCompanionDisplayAd(slot, bannerSizes, apesterElement);
+    }
+  }
+}
+
+/**
+ * @param {string} slot
+ * @param {Array} bannerSizes
+ * @param {AmpElement} apesterElement
+ * @return {Element}
+ */
+function constructCompanionDisplayAd(slot, bannerSizes, apesterElement) {
+  const defaultSize = bannerSizes[0];
+  const biggestAdSize = bannerSizes.reduce(
+    (maxSize, size) => {
+      maxSize.width = Math.max(maxSize.width, size[0]);
```

Comment:
I'm a bit confused there, this computes the max of both dimensions, but not max size. For example, if you have:

* 400x100
* 300x200

This will result in 400x200, which isn't any of the sizes defined. What is the intent here?

If the aspect ratios are the same, and this isn't a problem, then just checking the width or height should be sufficient. Either way, the intent here should be made clear.

If you want to get the larger of each dimension, create two separate reduces, or something like:

```javascript
const maxWidth = Math.max(...bannerSizes.map(s => s[0]));
const maxHeight = Math.max(...bannerSizes.map(s => s[1]));
```

which makes it very clear that you are just taking the larger of each dimension.

Reply:
OK, I did it that way for better performance, but I agree now is more readable so I'll change it.

Reply:
@sparhami  But we must add the   
// eslint-disable-next-line local/no-spread

Is that ok?

Reply:
We can't do spread operators.

We should do the following instead:

```js
const maxWidth = Math.max.apply(null, bannerSizes.map(s => s[0]));
const maxHeight = Math.max.apply(null, bannerSizes.map(s => s[1]));
```

Reply:
@alanorozco @sparhami fixed.

Reply:
Hi @alanorozco, 

@sparhami left AMP and says you are in charge now, can you take a look on the pr please?